modules/scaleform.lua

```diff
@@ -0,0 +1,208 @@
+---@class Scaleform : OxClass
+---@field name string
+---@field scaleform number
+---@field draw boolean
+---@field renderTarget number
+---@field targetName string
+---@field handle number
+local Scaleform = lib.class('Scaleform')
+
+---@param name string
+---@return nil
+---@description Create a new scaleform class
+function Scaleform:constructor(name)
+    self.name = name -- Set the name
+
+    local scaleform = lib.requestScaleformMovie(name) -- Request the scaleform movie
+    
+    if not scaleform then -- If the scaleform is nil
+        return error(('Failed to request scaleform movie - [%s]'):format(name)) -- Error the failed scaleform request
+    end
+
+    self.handle = scaleform -- Set the scaleform handle
+    self.draw = false -- Set the draw to false
+    self.fullScreen = true -- Set the full screen to false
+end
+
+---@param name string
+---@param args table
+---@return nil
+---@description Request a scaleform method with parameters
+function Scaleform:MethodArgs(name, args)
+    if not self.handle then -- If the scaleform handle is nil
+        return error('Scaleform handle is nil') -- Error the scaleform handle is nil
+    end
+    if type(args) ~= 'table' then -- If the type of args is not a table
+        return error('Args must be a table') -- Error args must be a table
+    end
+
+    BeginScaleformMovieMethod(self.handle, name) -- Begin the scaleform movie method
+    for i=1, #args do -- loop through the args
+        local arg = args[i] -- Set the value to the current arg
+        if type(arg) == 'string' then -- If the type of v is a string
+            ScaleformMovieMethodAddParamPlayerNameString(arg) -- Add the player name string
+        elseif type(arg) == 'number' then -- If the type of v is a number
+            if math.type(arg) == 'integer' then -- If the math type of v is an integer
+                ScaleformMovieMethodAddParamInt(arg) -- Add the integer
+            else -- If the math type of v is not an integer
+                ScaleformMovieMethodAddParamFloat(arg) -- Add the float
+            end
+        elseif type(arg) == 'boolean' then -- If the type of v is a boolean
+            ScaleformMovieMethodAddParamBool(arg) -- Add the boolean
+        else
+            error(('Unsupported Parameter type [%s]'):format(type(arg))) -- Error unsupported type
+        end
+    end
+    EndScaleformMovieMethod() -- End the scaleform movie method
+end
+
+---@param name string
+---@return nil
+---@description Request a scaleform method with no return value or parameters
+function Scaleform:Method(name)
```

Comment:
Why is there 3 different methods for running a scaleform method? Can we not merge it into one function that accepts a table of arguments, and decide what to do based on the arguments?

Reply:
Mainly for easier reading. so you can easily tell the difference between methods. however, method and method args could be merged together, Return would have to be seperate though, since it works completely differently

Reply:
From what I understand, it is technically possible to provide parameters *and* receive a return value. That's mainly my concern and why I want to merge the three.

Reply:
it would be quite messy to merge return into it and will 100% be better as a separate function, since the functions are different, adding args as a optional param to method could be good tho